sklearn/utils/validation.py

```diff
@@ -1020,6 +1021,147 @@ def check_scalar(x, name, target_type, min_val=None, max_val=None):
         raise ValueError('`{}`= {}, must be <= {}.'.format(name, x, max_val))
 
 
+def _check_psd_eigenvalues(lambdas, small_nonzeros_warning=True):
```

Comment:
@smarie please leave that default to false

Reply:
I will align of course to the majority vote but I do not agree with setting it to `False` here ; since it is already set to `False` at usage time in kernel PCA you should be happy with it, no ? [see here](https://github.com/scikit-learn/scikit-learn/pull/12145/files/13e2ba6de478ce6c3d2b59c84e1ab07eebdc51c1#diff-703280df0f9d49445c95026f6dbbee9fR216)

In the more general case of *other estimators* using the `_check_psd_eigenvalues` utility method, I think that this would be a better default to have it `True`. That way, people not aware of numerical imprecision or bad conditioning will be automaticallly aware about it. Let's not miss a chance to spot errors in the future !

Again, if the above arguments do not convince you I will modify it. Sorry for the over-engineering :)

Reply:
We currently only have one estimator using the function. It's using False. There's no need to address *potential* use-cases at the moment.

Reply:
Ok then, aligning with you and setting default to `False`.